### ISSUE TITLE
haproxy doesn't got SIGUSR1

### ISSUE THREAD

**Issue by julyclyde** [haproxy/haproxy#56]
I'm using service_loadbalancer from kubernetes ( https://github.com/kubernetes/contrib/tree/master/service-loadbalancer ) . This program would re-spawn haproxy when it found a change of upstream endpoints.
When service_loadbalancer starts, it runs haproxy -sf $(cat pidfile) several times very quickly, and on that moment, haproxy -sf doesn't take effect, there were many haproxy processes left.

I ran strace on the new haproxy -sf , and ensured it did send a SIGUSR1 to the old process indicated by pidfile.
I tried to send SIGUSR1, SIGTERM, SIGINT to the left haproxy processes while strace-ing on them, but they did not get the signals. I checkec /proc/PID/status file, the SigBlk line is fffffffc7bfa7a27. SIGKILL can kill the old process.

I searched SIG_SETMASK in the source code, and found it in signal.c file. Maybe you should remove SIGUSR1 from blocked_sig set in signal_init() function?

**Comment by julyclyde** [comment]
inspected old process with gdb:
(gdb) where
#0  0x00007f42b2916763 in __epoll_wait_nocancel () from /lib64/libc.so.6
#1  0x00000000004d47db in _do_poll (p=0x73a5e0 <cur_poller>, exp=1633798236) at src/ev_epoll.c:125
#2  0x000000000040a8ef in run_poll_loop () at src/haproxy.c:1576
#3  0x000000000040b438 in main (argc=8, argv=0x7fff4168bd08) at src/haproxy.c:1912
(gdb) print jobs
$1 = 4
(gdb) print stopping
$2 = 0

inspected old process with strace:
Process 46 attached
epoll_wait(0, {}, 200, 1000)            = 0
epoll_wait(0, {}, 200, 1000)            = 0
epoll_wait(0, {}, 200, 1000)            = 0
epoll_wait(0, {}, 200, 1000)            = 0
......

I don't know when did it block SIGUSR1 .

**Comment by wtarreau** [comment]
Issues reported here are not processed as this is *not* the official haproxy
project. Please post your problem to the mailing list instead. You don't need
to subscribe. Also please indicate your haproxy version and OS. I suspect you
may be with systemd as a few other such situations were reported already.

**Comment by julyclyde** [comment]
Thank you. I'll send this report to mailing list.

**Comment by julyclyde** [comment]
Sent to haproxy@formilux.org and got replied, but cannot see my mails in http://marc.info/?l=haproxy . What happened?

**Comment by wtarreau** [comment]
On Thu, Dec 10, 2015 at 06:34:39PM -0800, REN Xiaolei wrote:
> Sent to haproxy@formilux.org and got replied, but cannot see my mails in http://marc.info/?l=haproxy . What happened?

The mailing list server was listed on spamcop a few days ago, when this
happens it takes a few days for e-mails to show up again on marc.info/gmane.

**Comment by wtarreau** [comment]
@julyclyde, with your information and complements provided by Matti Savolainen I could implement and validate a workaround that I already merged into 1.7-dev :

   d50b4ac ("MEDIUM: unblock signals on startup.")

I'll backport this into 1.6 and 1.5 as well so that users affected by this issue can definitely get rid of it. I'm closing this issue now. Many thanks.
